library/std/src/sys/unix/locks/pthread_mutex.rs

```diff
@@ -113,6 +113,7 @@ impl Mutex {
         // Once it is used (locked/unlocked) or pthread_mutex_init() is called,
         // this behaviour no longer occurs.
         debug_assert!(r == 0 || r == libc::EINVAL);
```

Comment:
You shouldn't have both `debug_assert` and `assert` here. If were going to assert, you should delete the debug_assert.

I'm still hesitant about this one also, since it will be called from a drop impl as well.

Reply:
was just a copy and paste issue sorry about that! I do not think we need to move this in `assert` we have all the info inside r and we can check if it is right?